cfgmgr/buffermgr.cpp

```diff
@@ -194,20 +201,152 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port, string speed)
+ *  - BUFFER_QUEUE
+ *  - BUFFER_PORT_INGRESS_PROFILE_LIST
+ *  - BUFFER_PORT_EGRESS_PROFILE_LIST   
+ * One thing we need to handle is to transform the separator from | to :
+ * The following items contain separator:
+ *  - keys of each item
+ *  - pool in BUFFER_POOL
+ *  - profile in BUFFER_PG
+ */
+void BufferMgr::doBufferTableTask(Consumer &consumer, ProducerStateTable &applTable)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+        string key = kfvKey(t);
+
+        //transform the separator in key from "|" to ":"
+        transformSeperator(key);
+
+        string op = kfvOp(t);
+        if (op == SET_COMMAND)
+        {
+            vector<FieldValueTuple> fvVector;
+
+            SWSS_LOG_INFO("Inserting entry %s from CONFIG_DB to APPL_DB", key.c_str());
+
+            for (auto i : kfvFieldsValues(t))
+            {
+                SWSS_LOG_INFO("Inserting field %s value %s", fvField(i).c_str(), fvValue(i).c_str());
+                //transform the separator in values from "|" to ":"
+                if (fvField(i) == "pool")
+                    transformReference(fvValue(i));
+                if (fvField(i) == "profile")
+                    transformReference(fvValue(i));
+                if (fvField(i) == "profile_list")
+                    transformReference(fvValue(i));
+                fvVector.emplace_back(FieldValueTuple(fvField(i), fvValue(i)));
+                SWSS_LOG_INFO("Inserting field %s value %s", fvField(i).c_str(), fvValue(i).c_str());
+            }
+            applTable.set(key, fvVector);
+        }
+        else if (op == DEL_COMMAND)
+        {
+            SWSS_LOG_INFO("Removing entry %s from APPL_DB", key.c_str());
+            applTable.del(key);
+        }
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
 void BufferMgr::doTask(Consumer &consumer)
 {
     SWSS_LOG_ENTER();
 
     string table_name = consumer.getTableName();
 
+    if (table_name == CFG_BUFFER_POOL_TABLE_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferPoolTable);
+        return;
+    }
+
+    if (table_name == CFG_BUFFER_PROFILE_TABLE_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferProfileTable);
+        return;
+    }
+
+    if (table_name == CFG_BUFFER_PG_TABLE_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferPgTable);
+        return;
+    }
+
+    if (table_name == CFG_BUFFER_QUEUE_TABLE_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferQueueTable);
+        return;
+    }
+
+    if (table_name == CFG_BUFFER_PORT_INGRESS_PROFILE_LIST_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferIngressProfileListTable);
+        return;
+    }
+
+    if (table_name == CFG_BUFFER_PORT_EGRESS_PROFILE_LIST_NAME)
+    {
+        doBufferTableTask(consumer, m_applBufferEgressProfileListTable);
+        return;
+    }
+
```

Comment:
better to move this after the lossless pg profiles are generated to avoid the second copy during init phase

Reply:
I'm afraid that won't yield too much benefit because
- `doBufferTableTask` is called by `doTask` which is called for each **entry** in a table it listens to but not for the entire table.
- after a pg profile is generated, there will always be a call to `doTask` for the entry regardless of the order between pg profile generating and those calls to `doBufferTableTask`.